lib/logflare/backends/adaptor/postgres_adaptor/shared_repo.ex

```diff
@@ -0,0 +1,97 @@
+defmodule Logflare.Backends.Adaptor.PostgresAdaptor.SharedRepo do
+  use Ecto.Repo,
+    otp_app: :logflare,
+    adapter: Ecto.Adapters.Postgres
+
+  require Logger
+
+  alias Logflare.Source
+  alias Logflare.Backends.Backend
+
+  alias Logflare.Backends.Adaptor.PostgresAdaptor.Supervisor
+  alias Logflare.Backends.Adaptor.PostgresAdaptor.Repo.Migrations
+
+  def start(%Backend{} = backend) do
+    config = backend.config
+    pool_size = Application.get_env(:logflare, :postgres_backend_adapter)[:pool_size] || 10
+    schema = config["schema"] || config[:schema]
+
+    opts = [
+      url: config["url"] || config.url,
+      name: Supervisor.via(backend),
```

Comment:
is the via function for the repo or for the supervisor? if for the repo then shouldn't it be in this module?

Reply:
I did it in supervisor to hide the name of the `DynamicSupervisor` from the `Repo`, as it is irrelevant implementation detail for the `SharedRepo`.